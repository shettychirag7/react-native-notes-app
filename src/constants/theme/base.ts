import {Theme} from './type';
import {fonts} from '../fonts';

const base: Theme = {
  fontFamily: {
    titleLight: fonts.montserrat.light,
    titleRegular: fonts.montserrat.semiBold,
    titleBold: fonts.montserrat.bold,
    bodyLight: fonts.montserrat.thin,
    bodyBold: fonts.montserrat.medium,
    bodyRegular: fonts.montserrat.regular,
  },
  fontSize: {
    sm: 14,
    md: 16,
    lg: 20,
    xl: 24,
    xxl: 32,
  },
  space: {
    xs: 4,
    sm: 8,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 40,
  },
  colors: {
    primary: {
      10: '#aceef5',
      20: '#70e3f0',
      30: '#00d7ea',
      40: '#00cee6',
      50: '#00c4e1',
      60: '#00b4ce',
      70: '#009fb3',
      80: '#008b9a',
      90: '#00676c',
    },
    secondary: {
      10: '#f5f5f5',
      20: '#eeeeee',
      30: '#e0e0e0',
      40: '#bebebe',
      50: '#9f9f9f',
      60: '#767676',
      70: '#626262',
      80: '#424242',
      90: '#212121',
    },
    neutral: {
      N10: '#F7F7F7',
      N20: '#E1E1E1',
      N30: '#A8A8A8',
      N40: '#757575',
      N50: '#565656',
      N60: '#3B3B3B',
      N70: '#222222',
      N80: '#121212',
      white: '#FFFFFF',
      black: '#000000',
    },
  },
};

export default base;
