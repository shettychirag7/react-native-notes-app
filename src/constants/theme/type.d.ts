export type Theme = {
  fontFamily: {
    titleLight: string;
    titleRegular: string;
    titleBold: string;
    bodyLight: string;
    bodyBold: string;
    bodyRegular: string;
  };
  fontSize: {
    sm: number;
    md: number;
    lg: number;
    xl: number;
    xxl: number;
  };
  space: {
    xs: number;
    sm: number;
    md: number;
    lg: number;
    xl: number;
    xxl: number;
  };
  colors: {
    primary: {
      10: string;
      20: string;
      30: string;
      40: string;
      50: string;
      60: string;
      70: string;
      80: string;
      90: string;
    };
    secondary: {
      10: string;
      20: string;
      30: string;
      40: string;
      50: string;
      60: string;
      70: string;
      80: string;
      90: string;
    };
    neutral: {
      N10: string;
      N20: string;
      N30: string;
      N40: string;
      N50: string;
      N60: string;
      N70: string;
      N80: string;
      white: string;
      black: string;
    };
  };
};
