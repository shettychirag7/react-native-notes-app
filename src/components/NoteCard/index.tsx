import React from 'react';
import {TouchableOpacity, TouchableOpacityProps, View} from 'react-native';

import Text from '../Text';

import useStylePicker from './notecard.style';

export interface NoteCardProps extends TouchableOpacityProps {
  /**
   * title of note
   */
  title: string;
  /**
   * description of note
   */
  description: string;
  /**
   * Category type of notes.
   */
  category: string;
}

const NoteCard = (props: NoteCardProps) => {
  const {title, category, description, ...rest} = props;

  const styles = useStylePicker();

  return (
    <TouchableOpacity style={styles.cardview} {...rest}>
      <View>
        <Text type="cardTitle" style={styles.spacing}>
          {title}
        </Text>
        <Text
          type="body"
          style={styles.spacing}
          numberOfLines={3}
          ellipsizeMode={'tail'}>
          {description}
        </Text>
        <Text type="bodyLight" numberOfLines={1}>
          {category}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

export default NoteCard;
