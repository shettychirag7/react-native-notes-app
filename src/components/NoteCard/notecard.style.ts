import {StyleSheet} from 'react-native';

import {createStylePickerHook, GeneratorProps} from '../../hooks/style';

const OFFSET_MARGIN = 16;

const getStyleObject = ({theme, screen}: GeneratorProps) =>
  StyleSheet.create({
    cardview: {
      width: screen.width / 2 - OFFSET_MARGIN,
      alignItems: 'center',
      backgroundColor: theme.colors.neutral.white,
      borderRadius: theme.space.md,
      shadowColor: theme.colors.neutral.N30,
      shadowOffset: {width: 0, height: 0},
      shadowOpacity: 1,
      shadowRadius: theme.space.sm,
      elevation: theme.space.sm,
      flexDirection: 'row',
      justifyContent: 'space-between',
      margin: theme.space.sm,
      padding: theme.space.md,
    },
    spacing: {
      marginVertical: 4,
    },
    bottom: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      flexWrap: 'wrap',
    },
    bottomItem: {},
  });

const styles = createStylePickerHook(getStyleObject);

export default styles;
