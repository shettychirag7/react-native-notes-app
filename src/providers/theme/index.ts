import {createContext} from 'react';
import themes from '../../constants/theme';

export const ThemeContext = createContext({
  theme: themes.base,
});

export const ThemeProvider = ThemeContext.Provider;
