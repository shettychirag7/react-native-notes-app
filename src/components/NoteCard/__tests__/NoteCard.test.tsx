import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import NoteCard from '..';
import {ThemeProvider} from '../../../providers/theme';
import themes from '../../../constants/theme';

describe('NoteCard component tests', () => {
  it('renders NoteCard component', () => {
    const {toJSON} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <NoteCard
          title="Hello"
          description="Hello World"
          category="General"
          creationDate="12 Mar '22"
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('attaches onPress back', () => {
    const mockOnPress = jest.fn();
    const {getByTestId} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <NoteCard
          testID="note_card"
          title="Hello"
          description="Hello World"
          category="General"
          creationDate="12 Mar '22"
          onPress={mockOnPress}
        />
      </ThemeProvider>,
    );

    fireEvent.press(getByTestId('note_card'));

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
