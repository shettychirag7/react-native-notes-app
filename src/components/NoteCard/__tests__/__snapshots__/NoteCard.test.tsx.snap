// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCard component tests renders NoteCard component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 16,
      "elevation": 8,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 8,
      "opacity": 1,
      "padding": 16,
      "shadowColor": "#A8A8A8",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 8,
      "width": 359,
    }
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#222222",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 16,
            "letterSpacing": 0.1,
            "lineHeight": 21.6,
          },
          Object {
            "marginVertical": 4,
          },
        ]
      }
    >
      Hello
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={3}
      style={
        Array [
          Object {
            "color": "#222222",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 16,
          },
          Object {
            "marginVertical": 4,
          },
        ]
      }
    >
      Hello World
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#008b9a",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
          },
          undefined,
        ]
      }
    >
      General
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#008b9a",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
          },
          undefined,
        ]
      }
    >
      12 Mar '22
    </Text>
  </View>
</View>
`;
